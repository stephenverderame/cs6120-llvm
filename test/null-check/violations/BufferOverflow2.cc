#include <cassert>
#include <cstdio>
#include <cstdlib>
// NOLINTBEGIN
int main(int argc, const char** argv)
{
    int* vec = (int*)malloc(argc);
    assert(vec != nullptr);
    for (int i = 0; i < argc; ++i) {
        const auto t = argv + 1 + i;
        assert(t != nullptr);
        const auto word = *t;
        assert(word != nullptr);
        int num = atoi(word);
        vec[i] = num;
    }
}
// NOLINTEND