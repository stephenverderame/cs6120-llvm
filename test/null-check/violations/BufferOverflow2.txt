[31mSafety Violation[00m: Use of potentially null pointer in store at BufferOverflow2.cc:15
Because an index could not be proven to be in range, the following counterexample was found:
	[1margc[0m = 4
	[1mi[0m = 2
