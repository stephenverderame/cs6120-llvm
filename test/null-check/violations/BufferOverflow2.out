[31mSafety Violation[00m: Use of potentially null pointer in store at BufferOverflow2.cc:15
